tests/launcher.py

```diff
@@ -478,7 +539,139 @@ def make_mesh(self):
+            if specifics[0] == "'" and specifics[-1] == "'":
+                eosdcmd.extend(shlex.split(specifics[1:-1]))
+            else:
+                eosdcmd.extend(shlex.split(specifics))
+        eosdcmd.append('--config-dir')
+        eosdcmd.append(str(instance.config_dir_name))
+        eosdcmd.append('--data-dir')
+        eosdcmd.append(str(instance.data_dir_name))
+        eosdcmd.append('--genesis-json')
+        eosdcmd.append(f'{instance.config_dir_name}/genesis.json')
+        if self.args.timestamp:
+            eosdcmd.append('--genesis-timestamp')
+            eosdcmd.append(self.args.timestamp)
+
+        # Always enable a history query plugin on the bios node
+        if instance.name == 'bios':
+            if self.args.is_nodeos_v2:
+                eosdcmd.append('--plugin')
+                eosdcmd.append('eosio::history_api_plugin')
+                eosdcmd.append('--filter-on')
+                eosdcmd.append('"*"')
+            else:
+                eosdcmd.append('--plugin')
+                eosdcmd.append('eosio::trace_api_plugin')
+
+        if 'eosio::history_api_plugin' in eosdcmd and 'eosio::trace_api_plugin' in eosdcmd:
+            eosdcmd.remove('--trace-no-abis')
+            eosdcmd.remove('--trace-rpc-abi')
+            i = eosdcmd.index('eosio::trace_api_plugin')
+            eosdcmd.pop(i)
+            i -= 1
+            eosdcmd.pop(i)
+
+        if not instance.dont_start:
+            Utils.Print(f'spawning child: {" ".join(eosdcmd)}')
+
+            stdout = open(out, 'w')
+            stderr = open(err, 'w')
+            c = subprocess.Popen(eosdcmd, stdout=stdout, stderr=stderr)
+            with pidf.open('w') as pidout:
+                pidout.write(str(c.pid))
+            try:
+                err_sl.unlink()
+            except FileNotFoundError:
+                pass
+            err_sl.symlink_to(err.name)
+        else:
+            Utils.Print(f'unstarted node command: {" ".join(eosdcmd)}')
+
+            with open(instance.data_dir_name / 'start.cmd', 'w') as f:
+                f.write(' '.join(eosdcmd))
+
+    def bounce(self, nodeNumbers):
+        for num in nodeNumbers:
+            for node in self.network.nodes.values():
+                if self.network.name + num == node.name:
+                    with open(node.data_dir_name / f'{Utils.EosServerName}.pid', 'r') as f:
+                        pid = int(f.readline())
+                        self.terminate_wait_pid(pid, raise_if_missing=False)
+                    self.launch(node)
+
+    def down(self, nodeNumbers):
+        for num in nodeNumbers:
+            for node in self.network.nodes.values():
+                if self.network.name + num == node.name:
+                    with open(node.data_dir_name / f'{Utils.EosServerName}.pid', 'r') as f:
+                        pid = int(f.readline())
+                        self.terminate_wait_pid(pid)
```

Comment:
Would it be useful to have some debug logging in `bounce` and `down`?

Reply:
Added.